src/traces/pie/plot.js

```diff
@@ -97,7 +97,7 @@ function plot(gd, cdModule) {
                         pt.largeArc + (cw ? ' 1 ' : ' 0 ') + dx + ',' + dy;
                 }
 
-                var hole = trace.hole;
+                var hole = trace.hole[i] || trace.hole;
```

Comment:
To get the `hole` value or each slice you should use  `helpers.castOption` similar to the `pull` attribute.
```js
var hole = trace.hole;
if(hole) {
  hole = +helpers.castOption(trace.hole, pt.pts) || 0;
}
```

Reply:
Thanks, I've made that change :)